src/Mesh/Interpolation.jl

```diff
@@ -831,26 +827,31 @@ This interpolation function works for cubed spherical shell geometry.
  - `sv` State Array consisting of various variables on the discontinuous Galerkin grid
  - `v`  Interpolated variables
 """
-function interpolate_local!(intrp_cs::InterpolationCubedSphere{FT}, sv::AbstractArray{FT}, v::AbstractArray{FT}) where {FT <: AbstractFloat}
+function interpolate_local!(intrp_cs::InterpolationCubedSphere{FT}, sv::AbstractArray{FT}, v::AbstractArray{FT}; project=false) where {FT <: AbstractFloat}
     #------------------------------------------------------------------------------------------
     offset = intrp_cs.offset; m1_r = intrp_cs.m1_r; wb = intrp_cs.wb;
     ξ1 = intrp_cs.ξ1; ξ2 = intrp_cs.ξ2; ξ3 = intrp_cs.ξ3
     flg = intrp_cs.flg; fac = intrp_cs.fac
+    lati = intrp_cs.lati; longi = intrp_cs.longi
+    lat_grd = intrp_cs.lat_grd; long_grd = intrp_cs.long_grd
     #------------------------------------------------------------------------------------------
-    qm1   = length(m1_r)
-    nvars = size(sv,2)
-    Nel   = length(offset) - 1
+    qm1    = length(m1_r)
+    nvars  = size(sv,2)
+    Nel    = length(offset) - 1
+    np_tot = size(v,1) 
 
     device = typeof(sv) <: Array ? CPU() : CUDA()
 
-    if device==CPU()
+    if device == CPU()
         #------------------------------------------------------------------------------------------
         Nel   = length(offset) - 1
 
         vout    = zeros(FT,nvars) #FT(0)
         vout_ii = zeros(FT,nvars) #FT(0)
         vout_ij = zeros(FT,nvars) #FT(0)
 
+        _ρu, _ρv, _ρw = 2, 3, 4
```

Comment:
This is really specific to AtmosModel.

I think it would make more sense to put this into a separate function: ideally it would be integrated with the stuff in src/Diagnostics.

Reply:
Since this fixes a somewhat urgent problem, I'm happy to merge it in for the time being. We need to come up with a more robust solution though.